cmd/internal/cli/inspect_linux.go

```diff
@@ -257,132 +264,156 @@ var InspectCmd = &cobra.Command{
 	Example: docs.InspectExample,
 
 	Run: func(cmd *cobra.Command, args []string) {
-		// Sanity check
-		if _, err := os.Stat(args[0]); err != nil {
-			sylog.Fatalf("container not found: %s", err)
+		f, err := os.Stat(args[0])
+		if os.IsNotExist(err) {
+			sylog.Fatalf("Container not found: %s\n", err)
+		} else if err != nil {
+			sylog.Fatalf("Unable to stat file: %s", err)
 		}
+		sandboxImage := f.IsDir()
 
-		abspath, err := filepath.Abs(args[0])
-		if err != nil {
-			sylog.Fatalf("While determining absolute file path: %v", err)
+		var fimg sif.FileImage
+		if !sandboxImage {
+			var err error
+			fimg, err = sif.LoadContainer(args[0], true)
+			if err != nil {
+				sylog.Fatalf("failed to load SIF container file: %s", err)
+			}
+			defer fimg.UnloadContainer()
 		}
-		name := filepath.Base(abspath)
 
-		a := []string{"/bin/sh", "-c", ""}
+		var inspectData inspectFormat
+		inspectData.Type = containerType
+		inspectData.Data.Attributes.Labels = make(map[string]string, 1)
 
-		if listApps {
-			sylog.Debugf("Listing all apps in container")
-			a[2] += listAppsCommand
+		inspectShellCmd := []string{"/bin/sh", "-c", ""}
+
+		// Inspect the labels.
+		if (labels || defaultToLabels()) && AppName == "" {
```

Comment:
Do we need to do this labels check up here? or can we do it last like before so that we have simpler logic?

Reply:
It cant be like it was before, because if `--app` is not specified, then we need to check the label partition. If it is specified, then we need to shell/exec the container for the labels. We _can_ remove the `else if` part and just have a `if (labels || defaultToLabels()) && AppName != ""`, but that doesn't really make it more clear on whats going on.

I added some comments to make it a little more clearer.